### PR TITLE
feat: updating validation model

### PR DESCRIPTION
**Pull request by SamRobinson75684**
## Description

# updating validation model

## Type of changes

- [x] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [x] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [ ] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 1]
```diff
@@ -104,18 +104,25 @@ create table dbo.SCREENING_PROGRAMS (
...
-    NHS_NUMBER BIGINT not null,
-    DATE_CREATED DATETIME not null,
-    constraint PK_VALIDATION_EXCEPTION primary key (VALIDATION_EXCEPTION_ID)
+    NHS_NUMBER BIGINT NOT NULL,
+    DATE_CREATED DATETIME NOT NULL,
+    DATE_RESOLVED DATETIME NULL,
+    RULE_ID INT NOT NULL,
+    RULE_DESCRIPTION NVARCHAR(255) NOT NULL,
+    RULE_CONTENT NVARCHAR(MAX) NOT NULL,
+    CATEGORY NVARCHAR(100) NOT NULL,
```
Comment (stephhou) [inline application/CohortManager/Set-up/database/create_statement.sql:116]: Category should be int

File: application/CohortManager/Set-up/database/create_statement.sql [hunk 2]
```diff
@@ -104,18 +104,25 @@ create table dbo.SCREENING_PROGRAMS (
...
+    SCREENING_SERVICE NVARCHAR(100) NOT NULL,
```
Comment (stephhou) [inline application/CohortManager/Set-up/database/create_statement.sql:117]: Screening service should be an int too

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 3]
```diff
@@ -113,7 +113,7 @@ public async Task Run_Should_Return_BadRequest_And_Create_Exception_When_NhsNumb
 
         // Assert
         Assert.AreEqual(HttpStatusCode.BadRequest, result.StatusCode);
-        _callFunction.Verify(call => call.SendPost(It.Is<string>(s => s == "CreateValidationExceptionURL"), It.Is<string>(s => s.Contains("NhsNumberMustBeTenDigits"))), Times.Once());
+        _callFunction.Verify(call => call.SendPost(It.Is<string>(s => s == "CreateValidationExceptionURL"), It.Is<string>(s => s.Contains("1"))), Times.Once());
```
[the remark this answers is not in the document]
Comment (alex-clayton-1) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:116]: Can't you use the rule name as the RuleDescription like you have done for LookupValidation?

[the remark this answers is not in the document]
Comment (SamRobinson75684) [reply tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs]: good idea, I have changed it now

**Review by alex-clayton-1 — APPROVED** [review]

File: application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs [hunk 4]
```diff
@@ -25,9 +26,10 @@ public StaticValidation(ILogger< StaticValidation> logger, ICallFunction callFun
     [Function("StaticValidation")]
     public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Anonymous, "get", "post")] HttpRequestData req)
     {
-        var workflow = "Common";
-        ParticipantCsvRecord participantCsvRecord;
+        // Set which ruleset to use, needs to be parameterised
```
Comment (Warren-Pitterson) [inline application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs:29]: Can we remove the comments please

File: application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs [hunk 5]
```diff
@@ -25,9 +26,10 @@ public StaticValidation(ILogger< StaticValidation> logger, ICallFunction callFun
...
+        var screeningService = 1;
 
+        // Deserialisation
```
Comment (Warren-Pitterson) [inline application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs:32]: Can we remove the comments please

File: application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs [hunk 6]
```diff
@@ -36,45 +38,52 @@ public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Ano
...
 
-            participantCsvRecord = JsonSerializer.Deserialize<ParticipantCsvRecord>(requestBodyJson);
+            _participantCsvRecord = JsonSerializer.Deserialize<ParticipantCsvRecord>(requestBodyJson);
         }
         catch
         {
             return req.CreateResponse(HttpStatusCode.BadRequest);
         }
 
+        // Get rules
```
Comment (Warren-Pitterson) [inline application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs:48]: Can we remove the comments please

File: application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs [hunk 7]
```diff
@@ -36,45 +38,52 @@ public async Task<HttpResponseData> RunAsync([HttpTrigger(AuthorizationLevel.Ano
...
         string json = File.ReadAllText("staticRules.json");
         var rules = JsonSerializer.Deserialize<Workflow[]>(json);
 
-        var reSettings = new ReSettings{
+        var reSettings = new ReSettings
+        {
             CustomTypes = [typeof(Regex)]
         };
 
+        // Validation
```
Comment (Warren-Pitterson) [inline application/CohortManager/src/Functions/ScreeningValidationService/StaticValidation/StaticValidation.cs:57]: Can we remove the comments please

**Pull request merged**